src/dpms/dpmsWidget.tsx

```diff
@@ -469,30 +477,41 @@ fetching database name from fully qualified name structure */
           response
             .json()
             .then(async (responseResult: any) => {
-              const filteredEntries = responseResult.results.filter(
-                (entry: { displayName: string }) => entry.displayName
-              );
-              const databaseNames: string[] = [];
-              const updatedDatabaseDetails: { [key: string]: string } = {};
-              filteredEntries.forEach(
-                (entry: { description: string; displayName: string }) => {
-                  databaseNames.push(entry.displayName);
-                  const description = entry.description || 'None';
-                  updatedDatabaseDetails[entry.displayName] = description;
+              if (responseResult?.results) {
+                const filteredEntries = responseResult.results.filter(
+                  (entry: { displayName: string }) => entry.displayName
+                );
+                const databaseNames: string[] = [];
+                const updatedDatabaseDetails: { [key: string]: string } = {};
+                filteredEntries.forEach(
+                  (entry: { description: string; displayName: string }) => {
+                    databaseNames.push(entry.displayName);
+                    const description = entry.description || 'None';
+                    updatedDatabaseDetails[entry.displayName] = description;
+                  }
+                );
+                setDatabaseDetails(updatedDatabaseDetails);
+                setDatabaseNames(databaseNames);
+                setTotalDatabase(databaseNames.length);
+              } else {
+                if (responseResult?.error?.code) {
+                  setApiError(true);
```

Comment:
do we setApiError false anywhere? remove it on success?

Reply:
By default it will be false only for error it will be true and reload is enabled on changing notebooks so that the widget will load again

Reply:
resolved - handled ApiError in both conditions.